src/hotspot/os/linux/osContainer_linux.cpp

```diff
@@ -60,11 +61,18 @@ void OSContainer::init() {
   if (cgroup_subsystem == NULL) {
     return; // Required subsystem files not found or other error
   }
-  // We need to update the amount of physical memory now that
-  // cgroup subsystem files have been processed.
-  if ((mem_limit = cgroup_subsystem->memory_limit_in_bytes()) > 0) {
+  if ((mem_limit = cgroup_subsystem->memory_limit_in_bytes()) > 0 &&
+       mem_limit < host_memory) {
```

Comment:
I think the logic is not correct.

memory_limit_in_bytes() returns limit (0...x, but practically probably > 0) or -1 if unlimited or - I assume - if the limit could not be obtained.

So if -1, the second condition of the if branch holds true, and we set the physical memory to -1?

The code would be much clearer and more correct with
```
if (memlimit is set) {
  if (memlimit < host_memory) {
     set_physical_memory
  } else {
     log a warning
  }
}
```